pip/req/req_set.py

```diff
@@ -271,193 +307,196 @@ def prepare_files(self, finder):
-                # check that we don't already have an exact version match
-                # i.e. with at least one strict req operator
-                strict_req = set(('==', '===')) & set(
-                    op for op, _ in req_to_install.req.specs)
-                if self.upgrade and (not strict_req or
-                                     self.force_reinstall):
-                    if not (self.force_reinstall or req_to_install.link):
-                        try:
-                            link = finder.find_requirement(
-                                req_to_install, self.upgrade)
-                        except BestVersionAlreadyInstalled:
-                            best_installed = True
-                            install = False
-                        except DistributionNotFound as exc:
-                            not_found = exc
-                        else:
-                            # Avoid the need to call find_requirement again
-                            req_to_install.link = link
-
-                    if not best_installed:
-                        # don't uninstall conflict if user install and
-                        # conflict is not user install
-                        if not (self.use_user_site and not
-                                dist_in_usersite(
-                                    req_to_install.satisfied_by
-                                )):
-                            req_to_install.conflicts_with = \
-                                req_to_install.satisfied_by
-                        req_to_install.satisfied_by = None
-                else:
-                    install = False
-            if req_to_install.satisfied_by:
-                if best_installed:
-                    logger.info(
-                        'Requirement already up-to-date: %s',
-                        req_to_install,
-                    )
-                else:
-                    logger.info(
-                        'Requirement already satisfied (use --upgrade to '
-                        'upgrade): %s',
-                        req_to_install,
-                    )
+        # Tell user what we are doing for this requirement:
+        # obtain (editable), skipping, processing (local url), collecting
+        # (remote url or package name)
         if req_to_install.editable:
             logger.info('Obtaining %s', req_to_install)
-        elif install:
-            if (req_to_install.link and
-                    req_to_install.link.scheme == 'file'):
-                path = url_to_path(req_to_install.link.url)
-                logger.info('Processing %s', display_path(path))
+        else:
+            if not self.ignore_installed:
+                skip_reason = self._check_skip_installed(
+                    req_to_install, finder)
+            if req_to_install.satisfied_by:
+                logger.info(
+                    'Requirement already %s: %s', skip_reason,
```

Comment:
I think the conditional logic has changed here. Previously these logging statements would only ever happen if ``req_to_install.editable`` was ``False`` and ``req.ignore_installed`` was also ``False``, however now the logging statement will be attempted if ``req_to_install.editable`` is ``False`` only.

I think the

```python
if req_to_install.satisfied_by:
    skip_reason = self._check_skip_installed(req_to_install, finder)
if req_to_install.satisfied_by:
    logger.info(...)
```

was changed to

```python
if req_to_install.satisfied_by:
    skip_reason = self._check_skip_installed(req_to_install, finder)
    if req_to_install.satisfied_by:
        logger.info(...)
```

That would restore the old conditional logic. It would also fix a problem where ``skip_reason`` might be unset (If ``--ignore-installed`` is being used and the item is already installed).